Flick and Opposite Swipes Implemented

hsavoie opened the pull request:
1. Added Flick (users fling in all directions)

2. Added Swipe in opposite direction: This is just implemented by having the displayed text being randomized between two options.
_____________________________________________________

So SwipeInstruction.java has: 
   ```
    private String[] rightOptions = {"RIGHT", "OPPOSITE OF LEFT"};
    private String[] leftOptions = {"LEFT", "OPPOSITE OF RIGHT"};
    private String[] upOptions = {"UP", "OPPOSITE OF DOWN"};
    private String[] downOptions = {"DOWN", "OPPOSITE OF UP"};
```
    
Then for each case, a random number will determine which one of the two text options in the array will be displayed:

```
public void display() {
        rand2 = new Random();
        index = rand2.nextInt(rightOptions.length - 0) + 0;

        switch(currentAction){
                 case SWIPE_RIGHT:
                    displayTxt = rightOptions[index]; //either "RIGHT" or "OPPOSITE OF LEFT"
                    addTextView(displayTxt, SMALL_TEXT_SIZE);
                    break;
            case SWIPE_LEFT:
                        // and so on ......

```


Open to suggestions if either of you can think of a better way to do this :)

hsavoie committed:
Flick and Opposite Swipes Implemented

Added Flick (users fling in all directions)
Added Swipe in opposite direction:
      - This is just implemented by having the displayed text being randomized between two options (ex: "RIGHT" and "OPPOSITE OF LEFT")

fredy20501 commented on app/src/main/java/ca/unb/mobiledev/reflexrevolution/detectors/TouchDetector.java:
```diff
@@ -94,10 +95,14 @@ public boolean onTouch(View view, MotionEvent motionEvent) {
     }
     public enum SwipeAction {
-        SWIPE_RIGHT,
-        SWIPE_LEFT,
-        SWIPE_UP,
-        SWIPE_DOWN
+        FLICK(Type.FLING),
+        SWIPE_RIGHT(Type.SWIPE),
+        SWIPE_LEFT(Type.SWIPE),
+        SWIPE_UP(Type.SWIPE),
+        SWIPE_DOWN(Type.SWIPE);
```
We need to define the Type enum here as well. (Android studio should be giving you a bunch of errors because it doesn't know what `Type` is)
```java
public enum Type { FLING, SWIPE }
```

fredy20501 commented on app/src/main/java/ca/unb/mobiledev/reflexrevolution/instructions/SwipeInstruction.java:
```diff
@@ -12,6 +12,13 @@
     private final TouchDetector touchDetector;
     private TouchDetector.SwipeAction currentAction;
     private Random rand;
+    private Random rand2;
```
We don't need a separate random number generator, we can use the same one (it's random after all, so it doesn't change anything).

fredy20501 commented on app/src/main/java/ca/unb/mobiledev/reflexrevolution/instructions/SwipeInstruction.java:
```diff
@@ -12,6 +12,13 @@
     private Random rand;
+    private Random rand2;
+    private int index;
+    private String[] rightOptions = {"RIGHT", "OPPOSITE OF LEFT"};
+    private String[] leftOptions = {"LEFT", "OPPOSITE OF RIGHT"};
+    private String[] upOptions = {"UP", "OPPOSITE OF DOWN"};
+    private String[] downOptions = {"DOWN", "OPPOSITE OF UP"};
```
This works well. An possible improvement we could do is to merge all of it into 1 data structure instead of 4 arrays by using a HashMap. Something like this:
### Define HashMap
```java
private final Map<TouchDetector.SwipeAction, String[]> textLabels = new HashMap<>();
```
### Initialize values in `setup()`
Each array is associated with the enum value directly.
```java
// Add labels
textLabels.put(TouchDetector.SwipeAction.SWIPE_RIGHT, new String[] {"RIGHT", "OPPOSITE OF LEFT"});
textLabels.put(TouchDetector.SwipeAction.SWIPE_LEFT, new String[] {"LEFT", "OPPOSITE OF RIGHT"});
textLabels.put(TouchDetector.SwipeAction.SWIPE_UP, new String[] {"UP", "OPPOSITE OF DOWN"});
textLabels.put(TouchDetector.SwipeAction.SWIPE_DOWN, new String[] {"DOWN", "OPPOSITE OF UP"});
```
### Get value to display
Don't need a switch case anymore!
```java
@Override
public void display() {
    if (currentAction.getType() == TouchDetector.Type.SWIPE) {
        // Add small label for swipe instructions
        int index = rand.nextInt(2);
        String displayTxt = Objects.requireNonNull(textLabels.get(currentAction))[index];
        addTextView(displayTxt, SMALL_TEXT_SIZE);
        // Main label
        addTextView("SWIPE", DEFAULT_TEXT_SIZE);
    }
    else {
        // Main label
        addTextView("FLICK", DEFAULT_TEXT_SIZE);
    }
}
```

hsavoie committed:
Updated Reverse Swipe

- public enum Type {FLING, SWIPE} didnt get pushed last time PR so ensured it did this time
- improved the way the titles are being stored
- all other minor comments fixed

hsavoie committed:
Updated Reverse Swipe

Forgot to commit these changes

fredy20501 reviewed (approved):
Good job! 🏆

carterjmoore reviewed (approved):
👍 💯 🥇 🏆

carterjmoore committed:
Renamed Fling enum to Flick
